weechat-rs/src/buffer.rs

```diff
@@ -215,20 +215,24 @@ impl Nick {
     /// * `property` - The name of the property to get the value for, this can
     ///     be one of name, color, prefix or prefix_color. If a unknown
     ///     property is requested an empty string is returned.
-    pub fn get_string(&self, property: &str) -> &str {
+    pub fn get_string(&self, property: &str) -> Option<&str> {
         let weechat = self.get_weechat();
         let get_string = weechat.get().nicklist_nick_get_string.unwrap();
         let c_property = CString::new(property).unwrap();
         let value = unsafe {
             let ret = get_string(self.buf_ptr, self.ptr, c_property.as_ptr());
-            CStr::from_ptr(ret)
+
+            if ret.is_null() {
+                None
+            } else {
+                CStr::from_ptr(ret).to_str().unwrap_or_default()
```

Comment:
This needs to be wrapped in Some().

Reply:
Woops, probably should have checked it compiled. I'll fix that